homeassistant/components/weatherkit/__init__.py

```diff
@@ -28,6 +39,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         ),
     )
 
+    try:
+        await coordinator.update_supported_data_sets()
+    except (
+        WeatherKitApiClientAuthenticationError,
+        WeatherKitApiClientError,
+    ):
+        LOGGER.exception("Error initializing integration")
+        return False
```

Comment:
```suggestion
    except WeatherKitApiClientAuthenticationError:
        LOGGER.error("Authentication error initializing integration")
        return False
    except WeatherKitApiClientError as ex:
        raise ConfigEntryNotReady from ex
```

Reply:
Why do we return False instead of raise ConfigEntryNotReady?

Reply:
Authentication failures are not recoverable without reauth being implemented, but transient API errors should we try

Reply:
I was under the impression that `LOGGER.exception` logged exception information automatically. I looked at other uses and they didn't pass in the exception manually.

Reply:
It does but we don't want the noise of a whole traceback in the log as we already know why it failed.

Reply:
I see, thanks!